autoPyTorch/pipeline/components/setup/network/base_network.py

```diff
@@ -105,22 +107,35 @@ def predict(self, loader: torch.utils.data.DataLoader) -> torch.Tensor:
         """
         Performs batched prediction given a loader object
         """
-        assert self.network is not None
-        self.network.eval()
-
+        if len(self.network_snapshots) == 0:
+            assert self.network is not None
+            return self._predict(network=self.network.float(), loader=loader).cpu().numpy()
+        else:
+            # if there are network snapshots,
+            # take average of predictions of all snapshots
+            Y_snapshot_preds = list()
+
+            for network in self.network_snapshots:
+                Y_snapshot_preds.append(self._predict(network.float(), loader))
+            Y_snapshot_preds = torch.stack(Y_snapshot_preds)
+            assert isinstance(Y_snapshot_preds, torch.Tensor)
```

Comment:
Doing is instance is painful (computationally), do `typing.cast()`

Reply:
I tried that but it was still giving the same error.